Split AsyncNodeDeletionQueueInlines.h out of AsyncNodeDeletionQueue.h to reduce header includes in Document.h

rniwa opened the pull request:
#### 2eee4210a13e5d8bec43ebb8e02f4d2cf57667be
<pre>
Split AsyncNodeDeletionQueueInlines.h out of AsyncNodeDeletionQueue.h to reduce header includes in Document.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=293279">https://bugs.webkit.org/show_bug.cgi?id=293279</a>

Reviewed by Chris Dumez.

This PR extracts AsyncNodeDeletionQueueInlines.h out of AsyncNodeDeletionQueue.h so that we can reduce the number
of eventual header includes in Document.h.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/AsyncNodeDeletionQueue.h:
(WebCore::AsyncNodeDeletionQueue::addIfSubtreeSizeIsUnderLimit): Deleted.
(WebCore::AsyncNodeDeletionQueue::deleteNodesNow): Deleted.
(WebCore::AsyncNodeDeletionQueue::canNodeBeDeletedAsync): Deleted.
(WebCore::AsyncNodeDeletionQueue::isNodeLikelyLarge): Deleted.
* Source/WebCore/dom/AsyncNodeDeletionQueueInlines.h: Copied from Source/WebCore/dom/AsyncNodeDeletionQueue.h.
(WebCore::AsyncNodeDeletionQueue::addIfSubtreeSizeIsUnderLimit):
(WebCore::AsyncNodeDeletionQueue::deleteNodesNow):
(WebCore::AsyncNodeDeletionQueue::canNodeBeDeletedAsync):
(WebCore::AsyncNodeDeletionQueue::isNodeLikelyLarge):
* Source/WebCore/dom/ContainerNodeAlgorithms.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/page/MemoryRelease.cpp:
* Source/WebCore/page/Page.cpp:

Canonical link: <a href="https://commits.webkit.org/295177@main">https://commits.webkit.org/295177@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/adde86ec547daa260aa59444288c4b4f1ead05bf

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104229 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23933 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14271 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109429 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54897 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/106269 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24301 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32478 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79165 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107235 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18881 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94038 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59492 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/12076 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54257 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12135 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111811 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31385 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23135 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88186 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31750 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90227 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87848 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32738 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25849 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16935 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31314 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/36677 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31108 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34444 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32668 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/adde86ec547daa260aa59444288c4b4f1ead05bf. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45632

rniwa committed:
Split AsyncNodeDeletionQueueInlines.h out of AsyncNodeDeletionQueue.h to reduce header includes in Document.h
https://bugs.webkit.org/show_bug.cgi?id=293279

Reviewed by Chris Dumez.

This PR extracts AsyncNodeDeletionQueueInlines.h out of AsyncNodeDeletionQueue.h so that we can reduce the number
of eventual header includes in Document.h.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/dom/AsyncNodeDeletionQueue.h:
(WebCore::AsyncNodeDeletionQueue::addIfSubtreeSizeIsUnderLimit): Deleted.
(WebCore::AsyncNodeDeletionQueue::deleteNodesNow): Deleted.
(WebCore::AsyncNodeDeletionQueue::canNodeBeDeletedAsync): Deleted.
(WebCore::AsyncNodeDeletionQueue::isNodeLikelyLarge): Deleted.
* Source/WebCore/dom/AsyncNodeDeletionQueueInlines.h: Copied from Source/WebCore/dom/AsyncNodeDeletionQueue.h.
(WebCore::AsyncNodeDeletionQueue::addIfSubtreeSizeIsUnderLimit):
(WebCore::AsyncNodeDeletionQueue::deleteNodesNow):
(WebCore::AsyncNodeDeletionQueue::canNodeBeDeletedAsync):
(WebCore::AsyncNodeDeletionQueue::isNodeLikelyLarge):
* Source/WebCore/dom/ContainerNodeAlgorithms.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/page/MemoryRelease.cpp:
* Source/WebCore/page/Page.cpp:

Canonical link: https://commits.webkit.org/295177@main
